athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/handlers/GlueMetadataHandler.java

```diff
@@ -95,6 +97,10 @@
 {
     private static final Logger logger = LoggerFactory.getLogger(GlueMetadataHandler.class);
 
+    /**
+     * The maximum number of tables returned in a single response (as defined in the Glue API docs).
+     */
+    protected static final int GET_TABLES_REQUEST_MAX_RESULTS = 100;
```

Comment:
why do you need to hard code this? seems strange to me

Reply:
Glue has its own pagination. In order to satisfy the `pageSize` I needed to specify in the Glue Request the maximum results to retrieve (Glue allows `1-100`). Failure to do so may result in too many tables returned from Glue. If that happens, either we won't be able to satisfy the size contract in the **SDK**, or if we truncate the list (which **WS** will do), the `nextToken` from Glue will be invalidated by our logic. The solution is to retrieve the minimum between the `pageSize` and the max results that can be returned from Glue, and iterate until either the `nextToken` is null or the `pageSize` has been satisfied. Rather than hard-code the max results allowed in Glue, I opted for the above solution.

Reply:
lets remove this. getting a few extra rows (max 99) from Glue seems preferable to have this hard coded incase Glue's behavior ever changes in the future.